falcon/response.py

```diff
@@ -114,6 +132,12 @@ def __init__(self):
         self.stream = None
         self.stream_len = None
 
+        if self.context_type is None:
+            # PERF(kgriffs): The literal syntax is more efficient than dict().
+            self.context = {}
+        else:
+            self.context = self.context_type()
+
```

Comment:
self.context = {} if self.context_type is None or self.context_type() (more compact?)